src/js/models/maps/Map.js

```diff
@@ -258,6 +281,37 @@ define([
             if (isNonEmptyArray(config.terrains)) {
               this.set("terrains", new MapAssets(config.terrains));
             }
+
+            if (isNonEmptyArray(config.zoomPresets)) {
+              const zoomPresets = config.zoomPresets.map(zoomPresetObj => {
+                const allLayers = this.get('layers')?.models ?? this.get('layerCategories')
+                  .getMapAssets().map(assets => assets.models).flat();
+
+                const enabledLayerIds = [];
+                const enabledLayerLabels = [];
+                for (const layer of allLayers) {
+                  if (zoomPresetObj.layerIds.find(id => id === layer.get('layerId'))) {
+                    enabledLayerIds.push(layer.get('layerId'));
+                    enabledLayerLabels.push(layer.get('label'));
```

Comment:
I've still got to loop through all layers in selectZoomPreset because that's how I hide all currently visible layers and show the ones that need to be shown. I think this is simpler since I don't have to generate the list of IDs or labels later, as I'm already going through the zoom presets now